downloads/swan-lake-release-notes/swan-lake-beta3/RELEASE_NOTE.md

```diff
@@ -0,0 +1,780 @@
+
+### Improvements
+
+#### Improved Configurable Variables to Support XML Types Through TOML Syntax
+
+The `configurable` feature is improved to support variables with XML types through the TOML syntax.
+
+For example, if the XML-typed configurable variables are defined in the following way,
+
+``` ballerina
+configurable xml xmlVar = ?;
+```
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+xmlVar = "<book><name>Sherlock Holmes</name></book>"
+```
+
+#### Improved Configurable Variables to Support Additional Fields and Rest Fields for Records
+
+The `configurable` feature is improved to support additional fields and rest fields in record variables through the TOML syntax.
+
+For example, if a configurable variable with an open record type is defined in the following way,
+
+```ballerina
+type Person record {
+};
+
+configurable Person person = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+[person]
+intVal = 22
+floatVal = 22.33
+stringVal = "abc"
+arrVal = [1,2,3]
+mapVal.a = "a"
+mapVal.b = 123
+```
+
+The additional fields that are created from the TOML values will have the following types.
+
+* TOML Integer - `int`
+* TOML Float - `float`
+* TOML String - `string`
+* TOML Boolean - `boolean`
+* TOML Table - `map<anydata>`
+* TOML Table array - `map<anydata>[]`
+
+Similarly, if a configurable variable with a record type that contains a rest field is defined in the following way,
+
+```ballerina
+public type Numbers record {|
+   int ...;
+|};
+
+configurable Numbers numbers = ?;
+```
+
+the values can be provided in the `Config.toml` file as follows.
+
+
+```toml
+num1 = 11
+num2 = 26
+```
+
+#### Improved the Printed Error Stacktrace to Include the Cause
+
+The error stack trace has been improved to include the error cause locations. Stack frames of the wrapped error causes are also added to the stack trace.
+
+For the following example,
+
+```ballerina
+public function main() {
+    panic bar();
+}
+function bar() returns error {
+    return error("a", y());
+}
+function y() returns error {
+    return x();
+}
+function x() returns error {
+    return error("b");
+}
+```
+
+the expected stack trace will be as follows.
+
+```
+error: a
+at cause_location.0:bar(main.bal:6)
+cause_location.0:main(main.bal:2)
+cause: b
+at cause_location.0:x(main.bal:14)
+cause_location.0:y(main.bal:10)
+... 2 more
+```
+#### New Runtime Java APIs
+
+##### Invoke the Ballerina Object Method Asynchronously
+
+New Java Runtime APIs are introduced to execute the Ballerina object method from Java. The object method caller can decide whether to execute the object method sequentially or concurrently using the appropriate API.
+
+If the caller can ensure that the given object and object method is isolated and no data race is possible for the mutable state with given arguments, they can use the `invokeMethodAsyncConcurrently` method, or otherwise, the `invokeMethodAsyncSequentially` method.
```

Comment:
```suggestion
If the object and the method are `isolated` and the caller can ensure that there will be no data race with arguments that have mutable state, they can use the `invokeMethodAsyncConcurrently` method, or otherwise, the `invokeMethodAsyncSequentially` method.
```